File: CSharp/Project/Talk/Originals/SimpleSlice.cs
#region The Worst of Legacy Code: Forensic Development

#endregion

using System.Data.Common;
using Talk.Required;

namespace Talk.Originals
{
    public class SimpleSlice
    {
        public void EscalateProblems(Loan[] loans, DbConnection conn)
        {
            foreach (Loan loan in loans)
            {
                if (loan.IsOverDue() && (loan.GetAmount() > 5000))
                {
                    loan.SaveStatus("Critical", conn);
                }
                else if (loan.IsOverDue() && (loan.GetAmount() > 50))
                {
                    loan.SaveStatus("Escalated", conn);
                }
            }
        }
    }
}

File: EcmaScript/program/app/simpleSlice.js
module.exports = {
    escalateProblems: function(loans) {
        loans.forEach(loan => {
            if(loan.isOverDue() && (loan.getAmount() > 5000)) {
                loan.saveStatus('Critical');
            }
            else if (loan.isOverDue() && (loan.getAmount() > 50)) {
                loan.saveStatus('Escalated');
            }
        });
    }
};